kore/src/Kore/Step/Axiom/Matcher.hs

```diff
@@ -171,14 +183,50 @@ matchEqualHeads (Pair (DV_ _ dv1) (DV_ _ dv2)) =
     push (Pair dv1 dv2)
 matchEqualHeads _ = empty
 
+matchExists
+    :: (MatchingVariable variable, MonadUnify unifier)
+    => Pair (TermLike variable)
+    -> MaybeT (MatcherT variable unifier) ()
+matchExists (Pair (Exists_ _ variable1 term1) (Exists_ _ variable2 term2)) =
+    matchBinder (Binder variable1 term1) (Binder variable2 term2)
+matchExists _ = empty
+
+matchForall
+    :: (MatchingVariable variable, MonadUnify unifier)
+    => Pair (TermLike variable)
+    -> MaybeT (MatcherT variable unifier) ()
+matchForall (Pair (Forall_ _ variable1 term1) (Forall_ _ variable2 term2)) =
+    matchBinder (Binder variable1 term1) (Binder variable2 term2)
+matchForall _ = empty
+
+matchBinder
+    :: (MatchingVariable variable, MonadUnify unifier)
+    => Binder variable (TermLike variable)
+    -> Binder variable (TermLike variable)
+    -> MaybeT (MatcherT variable unifier) ()
+matchBinder (Binder variable1 term1) (Binder variable2 term2) = do
+    Monad.guard (sort1 == sort2)
+    refreshed1 <- refreshVariable variable1
```

Comment:
Something that is not obvious here is that `refreshVariable` generates a new variable that is globally unique, and that, because of that, bound variables are added to `avoiding` and never removed. Could you add a comment about that?